src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
```

Comment:
It looks like what happens is if a user leaves off the version key in the metadata.json file, puppet will throw a `MissingMetadata` ruby exception from puppets module loader.

```
2017-03-27 13:28:29,217 ERROR [qtp1400259539-33] [puppetserver] Puppet No version module metadata provided for ntp
... ruby stacktrace ...
```

And just leaves that module out of the response.

I'm trying to write some code to handle this, but it doesn't seem like ruby here can catch the exception that gets thrown. At first I tried catching `Puppet::Module::MissingMetadata` since that's what gets thrown, but then I simplified it to just be an `Exception` when I try to access `mod.version` but the same thing happens. Version itself is an instance variable...so I can't do the mod.version.nil? check because even if it's left out of the metadata.json it looks like it gets initialized.

[It looks like](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/module.rb#L62-L71) at the very least we can rely on name always being a variable we can have per module so this could read something like:

```ruby
```